lib/taskana-core/src/test/java/acceptance/task/UpdateManualPriorityWithSpiAccTest.java

```diff
@@ -0,0 +1,135 @@
+import pro.taskana.task.api.models.ObjectReference;
+import pro.taskana.task.api.models.Task;
+import pro.taskana.task.internal.builder.TaskBuilder;
+import pro.taskana.task.internal.jobs.helper.TaskUpdatePriorityWorker;
+import pro.taskana.workbasket.api.WorkbasketPermission;
+import pro.taskana.workbasket.api.WorkbasketService;
+import pro.taskana.workbasket.api.models.WorkbasketSummary;
+import pro.taskana.workbasket.internal.builder.WorkbasketAccessItemBuilder;
+
+@TaskanaIntegrationTest
+@WithServiceProvider(
+    serviceProviderInterface = PriorityServiceProvider.class,
+    serviceProviders = TestPriorityServiceProvider.class)
+class UpdateManualPriorityWithSpiAccTest {
+
+  @TaskanaInject TaskanaEngine taskanaEngine;
+  @TaskanaInject TaskService taskService;
+  @TaskanaInject ClassificationService classificationService;
+  @TaskanaInject WorkbasketService workbasketService;
+
+  ClassificationSummary defaultClassificationSummary;
+  WorkbasketSummary defaultWorkbasketSummary;
+  ObjectReference defaultObjectReference;
+  TaskUpdatePriorityWorker worker;
+
+  @WithAccessId(user = "businessadmin")
+  @BeforeAll
+  void setup() throws Exception {
+    worker = new TaskUpdatePriorityWorker(taskanaEngine);
+    defaultClassificationSummary =
+        defaultTestClassification().priority(9).buildAndStoreAsSummary(classificationService);
+    defaultWorkbasketSummary = defaultTestWorkbasket().buildAndStoreAsSummary(workbasketService);
+
+    WorkbasketAccessItemBuilder.newWorkbasketAccessItem()
+        .workbasketId(defaultWorkbasketSummary.getId())
+        .accessId("user-1-1")
+        .permission(WorkbasketPermission.OPEN)
+        .permission(WorkbasketPermission.READ)
+        .permission(WorkbasketPermission.APPEND)
+        .buildAndStore(workbasketService);
+    defaultObjectReference = defaultTestObjectReference().build();
+  }
+
+  @WithAccessId(user = "user-1-1")
+  @Test
+  void should_setPriorityToManualAndNotUpdateAccordingToSpi_When_ManualPriorityPositive()
+      throws Exception {
+    Task task = createDefaultTask().manualPriority(123).buildAndStore(taskService);
+    worker.executeBatch(List.of(task.getId()));
+    Task result = taskService.getTask(task.getId());
+    assertThat(result.getPriority()).isEqualTo(result.getManualPriority()).isEqualTo(123);
+  }
+
+  @WithAccessId(user = "user-1-1")
+  @Test
+  void should_setPriorityUsingSpi_When_ManualPriorityNegative() throws Exception {
+    Task task = createDefaultTask().manualPriority(-1).buildAndStore(taskService);
+    Task result = taskService.getTask(task.getId());
+    assertThat(result.getPriority()).isNotEqualTo(-1);
+  }
+
+  @WithAccessId(user = "user-1-1")
+  @Test
+  void should_UpdatePriorityAccordingToSpi_When_UpdatedTaskHasNegativeManualPriority()
+      throws Exception {
+    Task task = createDefaultTask().manualPriority(123).buildAndStore(taskService);
+    task.setManualPriority(-3);
+    taskService.updateTask(task);
```

Comment:
these are different tests, aren't they? Maybe we should add at least a dedicated assertion for each test?